File: src/HotelBooking.Domain/Entities/Booking.cs
using HotelBooking.Domain.Common;
using HotelBooking.Infrastructure.Identity;

namespace HotelBooking.Domain.Entities;

public class Booking : BaseEntity
{
    public Guid UserId { get; set; }
    public ApplicationUser User { get; set; } = default!;

    public Guid RoomId { get; set; }
    public Room Room { get; set; } = default!;

    public DateTime CheckInDate { get; set; }
    public DateTime CheckOutDate { get; set; }

    public decimal TotalPrice { get; set; }
}

File: web/js/core/models/booking.js
export class Booking {
    constructor({ id, userEmail, hotelName, roomNumber, checkInDate, checkOutDate, totalPrice }) {
        this.id = id;
        this.userEmail = userEmail ?? null;
        this.hotelName = hotelName ?? null;
        this.roomNumber = roomNumber ?? null;
        this.checkInDate = new Date(checkInDate);
        this.checkOutDate = new Date(checkOutDate);
        this.totalPrice = totalPrice;
    }
}